contrib/checkbox-provider-ce-oem/bin/parse_optee_test.py

```diff
@@ -2,32 +2,68 @@
 
 import json
 import argparse
+import subprocess as sp
+import shlex
+import re
+from checkbox_support.snap_utils.snapd import Snapd
+from checkbox_support.snap_utils.system import get_gadget_snap
 
 
-def parse_json_file(filename, filter=False):
-    with open(filename, 'r') as file:
+def parse_json_file(filepath, filter=False, xtest=None):
+    with open(filepath, 'r') as file:
         data = json.load(file)
     for test in data:
         if (filter and test['suite'] == 'pkcs11') or \
            (not filter and test['suite'] not in ['pkcs11']):
-            print_test_info(test)
+            print_test_info(test, xtest)
 
 
-def print_test_info(test):
+def print_test_info(test, xtest):
     print("suite: {}".format(test['suite']))
     print("test_id: {}".format(test['test_id']))
-    print("description: {}\n".format(test['test_description']))
+    print("description: {}".format(test['test_description']))
+    print("tool: {}\n".format(xtest))
+
+
+def look_up_xtest():
+    if Snapd().list("x-test"):
+        xtest = "x-test.xtest"
+    elif look_up_gadget() is not False:
+        xtest = look_up_gadget()
+    else:
+        xtest = None
+    return xtest
+
+
+def look_up_gadget():
+    gadget = get_gadget_snap()
+    cmd = "snap info {}".format(gadget)
+    gadget_info = sp.run(shlex.split(cmd),
+                         stdout=sp.PIPE,
+                         stderr=sp.PIPE,
+                         text=True,
+                         timeout=10)
+    xtest = "{}.xtest".format(gadget)
+    if re.findall(xtest, gadget_info.stdout):
+        return xtest
+    else:
+        return False
```

Comment:
```suggestion
    xtest = False
    snap = Snapd().list(gadget)
    if snap["apps"]:
        for app in snap["apps"]:
            if app["name"] == "xtest":
                xtest = ".".join([app["snap"], app["name"])
        xtest = ["{}.{}".format(name] for app in snap["apps"]]
    return xtest
```

Reply:
@stanley31huang I've pushed the fix. Thanks for the review.